### ISSUE TITLE
About the pretrained model ResNetVGGm1

### ISSUE THREAD

**Issue by happyflyhot** [visionml/pytracking#50]
Hello, thanks for your nice work! Can you tell me the source of vggmconv1 features used in your ResNetVGGm1 model? I failed to find the complete model of vgg-m with the input(mean and std) that is the same as resnet18. Thanks a lot if you could provide the source of the complete vgg-m model:)
sincerely~

**Comment by martin-danelljan** [comment]
Hi. If I remember correctly (was a while ago) I used [this model](https://github.com/Cadene/pretrained-models.pytorch/blob/master/pretrainedmodels/models/vggm.py) and then I made a script that re-normalizes the weights to fit the torchvision resnet input, and reorders the weights from BGR to RGB. I could probably find the script if you would have any use for it.

I ended up comparing the network with the output from MatConvNet VGG-m, and they were very similar.

**Comment by happyflyhot** [comment]
Thanks a lot~

**Comment by happyflyhot** [comment]
@martin-danelljan Hi, may you share that transform script with me? I failed to re-normalizes the weights myself...

**Comment by martin-danelljan** [comment]
Hey. Sorry for the delay. Here is the code that I used. You need to have [this repo](https://github.com/Cadene/pretrained-models.pytorch) in your python path.

```python
import os
import sys
import torch
import torchvision
import pretrainedmodels

env_path = os.path.join(os.path.dirname(__file__), '..')
if env_path not in sys.path:
    sys.path.append(env_path)


def main():
    resnet18 = torchvision.models.__dict__['resnet18'](pretrained=True)
    vggm = pretrainedmodels.models.vggm()

    resnet18.vggmconv1 = vggm.features[0]

    # Convert to RGB
    weight = resnet18.vggmconv1.weight.clone()
    resnet18.vggmconv1.weight[:,0,:,:] = weight[:,2,:,:]
    resnet18.vggmconv1.weight[:,2,:,:] = weight[:,0,:,:]

    # Normalize
    vgg_mean = torch.Tensor(vggm.mean[::-1])
    torch_std = torch.Tensor([0.229, 0.224, 0.225]).view(1,-1,1,1)
    resnet18.vggmconv1.weight[...] = resnet18.vggmconv1.weight * (torch_std*255)


    torch.save(resnet18.state_dict(), '../features/pretrained/resnet18_vggmconv1.pth')


if __name__ == '__main__':
    main()

**Comment by noobgrow** [comment]
the model vggm is no longer aviable,could please send me one?   
 'vggm': {
        'imagenet': 
            'url': 'http://data.lip6.fr/cadene/pretrainedmodels/vggm-786f2434.pth',

**Comment by noobgrow** [comment]
should be
https://data.lip6.fr/cadene/pretrainedmodels/vggm-786f2434.pth

**Comment by noobgrow** [comment]
model = VGGM(num_classes=1000)
        checkpoint = torch.load('vggm-786f2434.pth')
        for key in list(checkpoint.keys()):
            if 'classifier.' in key:
                checkpoint[key.replace('classifier.', 'classif.')] = checkpoint[key]
                del checkpoint[key]
        model.load_state_dict(checkpoint)

**Comment by noobgrow** [comment]
model = VGGM(num_classes=1000)
        checkpoint = torch.load('vggm-786f2434.pth')
        for key in list(checkpoint.keys()):
            if 'classifier.' in key:
                checkpoint[key.replace('classifier.', 'classif.')] = checkpoint[key]
                del checkpoint[key]
        model.load_state_dict(checkpoint)
